educe/stac/util/cmd/count_shapes.py

```diff
@@ -0,0 +1,136 @@
+# Author: Eric Kow
+# License: CeCILL-B (French BSD3)
+
+"""Count and display subgraphs matching shapes of interest
+
+"""
+
+from __future__ import print_function
+from collections import Counter
+
+from educe.util import (add_corpus_filters, concat_l, fields_without)
+import educe.stac.graph as stacgraph
+
+from ..args import (get_output_dir, read_corpus)
+from ..output import write_dot_graph
+
+
+NAME = 'count-shapes'
+
+
+def _outgoing(gra, node):
+    """Return the outgoing edges for a given node"""
+    return [e for e in gra.links(node)
+            if gra.is_relation(e) and gra.rel_links(e)[0] == node]
+
+def _maybe_lozenge(gra, node):
+    """Return (if applicable) lozenge nodes/edges starting from the given node
+
+    If the given node looks like the start of a lozenge, return all of
+    the nodes participating in the lozenge. If not, return None
+
```

Comment:
Could I bother you into explicitly documenting parameters? I think I get the type of gra right now but fear future me might not like it so much.